src/lib/plugins/world.js

```diff
@@ -237,23 +246,17 @@ module.exports.player = function (player, serv, settings) {
       .filter(([x, z]) => Math.abs(x - playerChunkX) > view || Math.abs(z - playerChunkZ) > view)
       .forEach(([x, z]) => player._unloadChunk(x, z))
 
-    return spiral([view * 2, view * 2])
+    const promises = []
+    spiral([view * 2, view * 2])
       .map(t => ({
         chunkX: playerChunkX + t[0] - view,
         chunkZ: playerChunkZ + t[1] - view
       }))
       .filter(({ chunkX, chunkZ }) => serv._worldLoadPlayerChunk(chunkX, chunkZ, player))
-      .reduce((acc, { chunkX, chunkZ }) => {
-        const p = acc
-          .then(() => player.world.getColumn(chunkX, chunkZ))
-          .then((column) => player.sendChunk(chunkX, chunkZ, column))
-        return group ? p.then(() => sleep(5)) : p
```

Comment:
Removing this logic will cause nmp server to kill the vanilla client by sending chunks to fast.

Reply:
Ah i see, reverted. It seemed to not be waiting for all the promises to resolve before spawning the player into the world causing it to sometimes fall through the world but that may be something else